omegaconf/basecontainer.py

```diff
@@ -88,39 +88,41 @@ def __repr__(self) -> str:
 
     # Support pickle
     def __getstate__(self) -> Dict[str, Any]:
-        from omegaconf import DictConfig
+        from omegaconf import DictConfig, _utils
 
         dict_ = copy.copy(self.__dict__)
-        if sys.version_info < (3, 7, 0):  # pragma: no cover
-            ref_type = dict_["_metadata"].ref_type
-            if isinstance(self, DictConfig):
-                if ref_type is not Any:
-                    dict_["_metadata"].ref_type = Dict
-            else:
-                if ref_type is not Any:
-                    dict_["_metadata"].ref_type = List
+        ref_type = dict_["_metadata"].ref_type
+        if isinstance(self, DictConfig):
+            if ref_type is not Any:
+                key_type, element_type = _utils.get_dict_key_value_types(ref_type)
+                print(ref_type, key_type, element_type)
```

Comment:
You are using print way too much. :)
I do recommend that you install pycharm and learn to use the IDE debugger.

Reply:
When i know where is the problem i tend to use prints because I find it faster.